scene/gui/tree.cpp

```diff
@@ -1313,6 +1313,44 @@ bool TreeItem::is_button_disabled(int p_column, int p_index) const {
 	return cells[p_column].buttons[p_index].disabled;
 }
 
+void TreeItem::move_column(int p_column, int p_index) {
+	TreeItem::Cell tmp1 = cells[p_index];
+	cells.write[p_index] = cells[p_column];
+
+	if (p_column < p_index) {
+		for (int i = p_index - 1; i >= p_column; i--) {
+			TreeItem::Cell tmp2 = cells[i];
+			cells.write[i] = tmp1;
+			tmp1 = tmp2;
+		}
+	} else {
+		for (int i = p_index + 1; i <= p_column; i++) {
+			TreeItem::Cell tmp2 = cells[i];
+			cells.write[i] = tmp1;
+			tmp1 = tmp2;
+		}
+	}
+
+	for (int i = 0; i < get_child_count(); i++) {
+		get_child(i)->move_column(p_column, p_index);
+	}
+
+	_changed_notify(p_column);
+}
+
+void TreeItem::swap_columns(int p_column1, int p_column2) {
+	TreeItem::Cell tmp = cells[p_column1];
+	cells.write[p_column1] = cells[p_column2];
+	cells.write[p_column2] = tmp;
```

Comment:
There is a SWAP macro, which might be able to achieve the same.

Reply:
The cells vector uses the write method, which is incompatible with the SWAP macro, therefore i did not alter it.